recipes-extended/onl/onl.inc

```diff
@@ -36,6 +36,30 @@ PACKAGE_ARCH = "${MACHINE_ARCH}"
 PROVIDES += "libonlp libonlp-platform libonlp-platform-defaults"
 INSANE_SKIP:${PN} = "file-rdeps"
 
+PACKAGES_DYNAMIC = "^libonlp-.*"
+
+python populate_packages:prepend () {
+    # ONL uses dashes in the ARCH name
+    arch=d.getVar('TRANSLATED_TARGET_ARCH')
+    name_prefix='libonlp-' + arch
+
+    # prevent Yocto from being "helpful" and renaming packages based on the
+    # included lib's filename
+    def debian_noautoname(file, pkg, pattern, format, basename):
+        d.setVar('DEBIAN_NOAUTONAME:%s' % pkg, "1")
+
+    # only split out platform libraries (having arch in their name)
+    do_split_packages(
+        d, '${libdir}', name_prefix + '-(.*).so.1', name_prefix + '-%s',
+        'libonlp (' + arch + '-%s)', prepend=True, hook=debian_noautoname)
+    # create -dev packages as well for the .so symlink, else onl-dev will depend
+    # on everything
+    do_split_packages(
+        d, '${libdir}', name_prefix + '-(.*).so$', name_prefix + '-%s-dev',
+        'libonlp (' + arch + '-%s) - Development files', prepend=True,
+        allow_links=True)
+}
```

Comment:
> I am not sure I understand how this is useful now that we lump everything together into generic images.

The idea is to have a subset of platform support as built (or buildable) along side the image, without including it. For which we need to split it out of the onl package.

And the easiest way to do that is to create dynamic libonlp-* packages, like we do for the kernel modules.

> How do we verify that each packagegroup contains the correct kernel modules for the respective platforms?

The same way we (used to) verify that the correct kernel modules are in the appropriate machine config ;-).

Cross-referencing with the (ONL) platform init module which kernel modules are loaded. Though for most platforms this is rather obvious, since their modules have the platform name in the filename.